app/src/main/java/com/example/plantsapp/presentation/ui/plants/PlantsFragment.kt

```diff
@@ -11,14 +12,19 @@ import com.example.plantsapp.data.PlantsRepositoryImpl
 import com.example.plantsapp.presentation.ui.plants.adapter.PlantsAdapter
 import com.example.plantsapp.databinding.FragmentPlantsBinding
 import com.example.plantsapp.presentation.ui.plantcreation.PlantCreationFragment
+import com.example.plantsapp.presentation.ui.plantdetail.DetailViewModelFactory
 import com.example.plantsapp.presentation.ui.plantdetail.PlantDetailFragment
+import com.example.plantsapp.presentation.ui.plantdetail.PlantDetailViewModel
 
 class PlantsFragment : Fragment(R.layout.fragment_plants) {
 
     private val binding: FragmentPlantsBinding by viewBinding(FragmentPlantsBinding::bind)
     private val plantsViewModel: PlantsViewModel by viewModels {
         PlantsViewModelFactory(repository = PlantsRepositoryImpl)
     }
+    private val detailViewModel: PlantDetailViewModel by activityViewModels {
+        DetailViewModelFactory()
+    }
```

Comment:
Same here - your plant should not know anything about some "PlantDetailViewModel", because maybe we will decide to open something else instead of DetailScreen? We will need to change this logic too